internal/status/gateway.go

```diff
@@ -1,9 +1,51 @@
 package status
 
-import gwapiv1b1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+import (
+	corev1 "k8s.io/api/core/v1"
+	gwapiv1b1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 
-// SetGatewayScheduled adds or updates the Scheduled condition for the provided Gateway.
-func SetGatewayScheduled(gw *gwapiv1b1.Gateway, scheduled bool) *gwapiv1b1.Gateway {
-	gw.Status.Conditions = mergeConditions(gw.Status.Conditions, computeGatewayScheduledCondition(gw, scheduled))
+	"github.com/envoyproxy/gateway/internal/gatewayapi"
+)
+
+// SetGatewayStatus adds or updates status for the provided Gateway.
+func SetGatewayStatus(gw *gwapiv1b1.Gateway, scheduled bool, svc *corev1.Service) *gwapiv1b1.Gateway {
+	computeGatewayStatusAddrs(gw, svc)
+	gw.Status.Conditions = mergeConditions(gw.Status.Conditions,
+		computeGatewayScheduledCondition(gw, scheduled), computeGatewayReadyCondition(gw))
 	return gw
 }
+
+// computeGatewayStatusAddrs computes status addresses for the provided gateway
+// based on the status IP/Hostname of svc.
+func computeGatewayStatusAddrs(gw *gwapiv1b1.Gateway, svc *corev1.Service) {
+	var ingAddrs, ingHosts []string
```

Comment:
curious what the `ing` prefix is for ?

Reply:
It's short of `ingress`, since the addresses/hostnames are from `service.status.ingresses[0].ip`. I've removed `ing` from the var names to avoid confusion.